tests/test_on_data/ecephys/test_recording_interfaces.py

```diff
@@ -268,6 +267,29 @@ def test_devices_written_correctly(self, setup_interface):
 
         nwbfile.devices["Intan"].description == "RHD Recording System"
 
+    def test_not_adding_extra_devices_when_recording_has_groups(self, setup_interface):
+        # Test that no extra-devices are added when the recording has groups
+
+        nwbfile = mock_NWBFile()
+        self.interface.add_to_nwbfile(nwbfile=nwbfile)
+        recording = self.interface.recording_extractor
+        num_channels = recording.get_num_channels()
+        channel_groups = np.full(shape=num_channels, fill_value=0, dtype=int)
+        channel_groups[::2] = 1
+        recording.set_channel_groups(groups=channel_groups)
```

Comment:
Just curious, why is `self.interface.add_to_nwbfile(nwbfile=nwbfile)` not called after `recording.set_channel_groups(groups=channel_groups)`? Does the mock extractor already have channel groups when you add to the file?

Reply:
Thanks, the tests were wrong. Adding the interfaces' data to the nwbfile should be done after the setup. Should be corrected now.